wemake_python_styleguide/visitors/ast/statements.py

```diff
@@ -498,3 +508,108 @@ def _check_tuple_arguments_types(
             if isinstance(arg, self._no_tuples_collections):
                 self.add_violation(NotATupleArgumentViolation(node))
                 break
+
+
+@final
+class UnsafeGeneratorExpressionVisitor(BaseNodeVisitor):
+    """Checks if a generator expression is safe."""
+
+    def __init__(self, *args, **kwargs) -> None:
+        """Creates the lists that would hold the variables."""
+        super().__init__(*args, **kwargs)
+        self.inside_variables: List[str] = []
+        self.created_variables: List[str] = []
+        self.affecting_variables: List[str] = []
+
+    def visit(self, node: ast.AST) -> None:
+        """Checks if the generator expression has affecting variables."""
+        assignments = [
+            sub_node
+            for sub_node in ast.walk(node)
+            if isinstance(sub_node, ast.Assign)
```

Comment:
You only check for `Assign`, what about `AugAssign` like `+= 1` or assigning expression like `(x := 1)`? We need to support them as well.

Reply:
Sure, it needs to have AugAssign support, but it'll add too much complexity with the current Visitor.

In the case of the assigning expression, there is a Violation that prevents the usage of `(x := 1)` , WPS332 Found Walrus operator